docs: reformat the documentation

Revathyvenugopal162 opened the pull request:
fix #373 
This pull request enhances the documentation for the Ansys Sphinx theme with the following updates:

- User Guide: Updated to include comprehensive information.
- Examples: Added demonstrations of style changes for all applicable extensions.
- Copyright: Ensured proper attribution for examples, particularly from Sphinx Design and Admonitions, which were directly downloaded.
![exa_2](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/fabcae40-3b84-409e-89cb-ae5050a8455e)
![exam_3](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/bfc4f05d-20a8-4aef-bcfd-41180239f7cc)
![exam_4](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/a7f12db0-2a14-4a38-ac7d-3d44090416f7)
![exam_5](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/6d7a8562-50ad-4d8e-8bd3-b4f0f02fa539)
![exam_6](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/8bc85cfb-5b12-4685-9a1b-7e8b7f532f56)
![examples_1](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/cabba728-236a-43d3-8e21-e5fbfe071fbe)
![userguide](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/c7a42ea5-741f-46e6-9684-461d09c613ee)

jorgepiloto commented on doc/Makefile:
```diff
@@ -6,22 +6,27 @@ SPHINXOPTS    =
+# customized clean due to api examples and sphinx-gallery
 clean:
 	rm -rf $(BUILDDIR)
 	find . -type d -name "_autosummary" -exec rm -rf {} +
```
Maybe this one is no longer required...

jorgepiloto commented on doc/source/conf.py:
```diff
@@ -172,10 +190,12 @@ def extract_example_links(
         List of example links.
     """
-    g = Github()
+    access_token = os.getenv("GITHUB_ACCESS_TOKEN")
+    g = Github(access_token)
```
Do we really need a token for this?

Revathyvenugopal162 replied:
no, i needed it locally since i build a couple of times, i am removing it.

jorgepiloto commented on doc/source/examples/nbsphinx/jupyter-notebook.ipynb:
We may have to install PyVista with `trame` for this notebook.

jorgepiloto commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,78 @@
+
+From ansys sphinx theme you can use the following features:
+
+1. PyAnsys and Ansys logos
```
Let's avoid numbering these subsections. It is annoying to maintain if changing the order at some point.

jorgepiloto commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,78 @@
+
+``favicon`` is the icon that appears in the browser tab.
+
+#. **Version switcher**
```
This is getting rendering inside of previous subsection rather than being a subsection on its own.

Revathyvenugopal162 replied:
added

jorgepiloto commented on doc/styles/config/vocabularies/ANSYS/accept.txt:
```diff
@@ -9,8 +9,14 @@ link_code_library
 link_code_branch
 html_context
-autoapi
+(?i)autoapi
```
Just curious about what this syntax means.

SMoraisAnsys replied:
This is related to case-insensitivity, i.e. the matching should be case-insensitive.

jorgepiloto commented on pyproject.toml:
```diff
@@ -42,6 +42,13 @@ doc = [
     "PyGitHub==2.3.0",
+    "sphinx-gallery==0.16.0",
+    "Pillow>=9.0",
+    "pyvista==0.43.9",
```
```suggestion
    "pyvista[jupyter]==0.43.9",
```

SMoraisAnsys reviewed (commented):
Great job ! I left a few minor comments

jorgepiloto commented:
Also, could we have the same example for both `nbsphinx` and `sphinx-gallery`? I will add another one for the `jupytext` approach.

SMoraisAnsys commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+        extensions = [
+            "sphinx_gallery.gen_gallery",
+        ]
+"""
```
```suggestion
# sphinx_gallery_thumbnail_path = '_static/pyansys_light_square.png'

"""
```
Open question on my side. Since we added a thumbnail with `nbsphinx`, why not doing it with `sphinx-gallery` ?
If you agree with that, the proposed change should do the job

Revathyvenugopal162 replied:
i added in conf.py file

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+Examples
+########
+
+This show how the different extensions and documentation components are rendered in the ``ansys-sphinx-theme``.
```
```suggestion
This section provides examples that show how the Ansys Sphinx Theme renders different extensions and documentation components.
```
Use the "p;roduct" name rather than the package name for better readability.

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+Sphinx extensions
+=================
+
+These examples show how the different sphinx extensions are rendered in the ``ansys-sphinx-theme``.
```
```suggestion
These examples show how the Ansys Sphinx Theme renders different Sphinx extensions.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+      :link: sphinx-design
+      :link-type: ref
+
+      Examples of the sphinx design rendered using the `ansys-sphinx-theme`.
```
```suggestion
      Examples of how the Sphinx design is rendered.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+      :link: nbsphinx
+      :link-type: ref
+
+      Examples of notebooks rendered using the `ansys-sphinx-theme`.
```
```suggestion
      Examples of how notebooks are rendered.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+      :link: examples/gallery-examples/index
+      :link-type: doc
+
+      Examples of sphinx gallery rendered using the `ansys-sphinx-theme`.
```
```suggestion
      Examples of how the Sphinx gallery is rendered.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+      :link: sphinx-autoapi
+      :link-type: ref
+
+      Examples of autoapi rendered using the `ansys-sphinx-theme`.
```
```suggestion
      Examples of how Sphinx AutoAPI documentation is rendered.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+Documentation components
+========================
+
+These examples show how the different documentation components are rendered in the ``ansys-sphinx-theme``.
```
```suggestion
These examples show how the Ansys Sphinx Theme renders different documentation components.
```

PipKat commented on doc/source/examples.rst:
```diff
@@ -0,0 +1,81 @@
+      :link: admonitions
+      :link-type: ref
+
+      Examples of admonitions rendered using the `ansys-sphinx-theme`.
```
```suggestion
      Examples of how admonitions are rendered.
```

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+===========
+
+Admonitions are a way to highlight important information in documentation.
+This page shows how the admonitions are rendered in the documentation with ``ansys sphinx theme``.
```
```suggestion
This page shows how the Ansys Sphinx Theme renders admonitions.
```

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+Admonitions
+===========
+
+Admonitions are a way to highlight important information in documentation.
```
```suggestion
Admonitions are specially formatted blocks that highlight important information in the documentation.
```

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+
+    {% endfor %}
+
+Credits and copyright
```
Does this section and child section need to exist. It seems that the information can be provided as part of the Admonitions section, without headings?

Revathyvenugopal162 replied:
sure, we can add that

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+---------------------
+
+The examples shown above are derived from the PyData Sphinx Theme project, copyrighted by pandas, 2018.
+Original examples can be found at `PyData sphinx theme examples <pydata_examples_>`_.
```
```suggestion
Original examples can be found in `Examples <pydata_examples_>`_ in
the PyData Sphinx Theme documentation.
```

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+Original copyright
+~~~~~~~~~~~~~~~~~~
+
+The examples included here were originally copyrighted and licensed as follows:
```
The "included here" is referring to the examples in the PyData Sphinx Theme documentation or in the Ansys Sphinx Theme documentation? Again, I'm not sure why this can't be a paragraph after the examples rather than in multiple sections.

Revathyvenugopal162 replied:
examples included in ansys sphinx theme this page.

Revathyvenugopal162 replied:
added as `The examples included below in this page were originally copyrighted and licensed as follows:`

Revathyvenugopal162 replied:
and moved this sections on the sentence along with admonitions

PipKat commented on doc/source/examples/admonitions.rst:
```diff
@@ -0,0 +1,31 @@
+The examples included here were originally copyrighted and licensed as follows:
+
+- Copyright (c) 2021, Pradyun Gedam. Licensed under the Creative Commons Attribution-ShareAlike 4.0 International License
+  (SPDX-License-Identifier: CC-BY-SA-4.0), as specified by PyData Sphinx theme in their examples.
```
```suggestion
  (SPDX-License-Identifier: CC-BY-SA-4.0), as specified by the PyData Sphinx Theme in their examples.
```

PipKat commented on doc/source/examples/nbsphinx.rst:
```diff
@@ -0,0 +1,8 @@
+
+Jupyter notebook - nbsphinx
+===========================
+
```
Add a brief description to introduce the notebook?

Revathyvenugopal162 replied:
sure, added `This example shows how the Ansys Sphinx Theme renders a Jupyter notebook using the ``nbspinx`` extension.`

PipKat commented on doc/source/examples/nbsphinx/jupyter-notebook.ipynb:
```diff
@@ -0,0 +1,115 @@
+   "metadata": {},
+   "source": [
+    "# Nbsphinx example\n",
+    "This is an example for rendering of nbsphinx with ansys-sphinx-theme."
```
```suggestion
    "This example renders a Jupyter notebook using the ``nbsphinx`` extension."
```

PipKat commented on doc/source/examples/nbsphinx/jupyter-notebook.ipynb:
```diff
@@ -0,0 +1,115 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Rending equations using Math."
```
```suggestion
    "## Render equations using ``math`` from IPython."
```

PipKat commented on doc/source/examples/nbsphinx/jupyter-notebook.ipynb:
```diff
@@ -0,0 +1,115 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Rendering a table in markdown."
```
```suggestion
    "## Render a table in markdown."
```

PipKat commented on doc/source/examples/sphinx-autoapi.rst:
```diff
@@ -0,0 +1,13 @@
+.. _sphinx-autoapi:
+
+Sphinx autoapi
```
```suggestion
Sphinx AutoAPI
```

PipKat commented on doc/source/examples/sphinx-autoapi.rst:
```diff
@@ -0,0 +1,13 @@
+Sphinx autoapi
+==============
+
+This section describes ansys-sphinx-theme endpoints, their capabilities, and how
+to interact with them programmatically.
```
```suggestion
This section describes how to use Python to interact programmatically with the Ansys Sphinx Theme.
```
I'm not sure this is what you want for the introduction. It seems like it should be about Sphinx AutoAPI--not our theme?

PipKat commented on doc/source/examples/sphinx-design.rst:
```diff
@@ -4,7 +4,15 @@ Sphinx design
 =============
 The rendering of sphinx design with ansys sphinx theme.
 To access the full documentation for the sphinx design package,
```
```suggestion
This example shows how the Ansys Sphinx Theme renders documentation components
using the ``sphinx-design`` extension. For more information, see the
```

PipKat commented on doc/source/examples/sphinx-design.rst:
```diff
@@ -4,7 +4,15 @@ Sphinx design
 The rendering of sphinx design with ansys sphinx theme.
 To access the full documentation for the sphinx design package,
-please refer `sphinx design <https://sphinx-design.readthedocs.io/en/latest/index.html>`_.
+please refer `sphinx design <sphinx_design_docs_>`_.
```
```suggestion
`documentation <sphinx_design_docs_>`_.
```

PipKat commented on doc/source/examples/sphinx-gallery/README.rst:
```diff
@@ -0,0 +1,7 @@
+Sphinx gallery
+==============
+
+This is an example of a gallery generated with `Sphinx gallery`` and its
+rendering with ansys sphinx theme.
```
```suggestion
This example shows how the Ansys Sphinx Theme renders examples
using Sphinx Gallery. 
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+
+# noqa: D205, D400
+"""
+Sphinx gallery
```
```suggestion
Sphinx Gallery
```
It is a proper name, so add it to the accept.txt file for Vale.

PipKat commented on doc/source/examples/sphinx-gallery/README.rst:
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+Sphinx gallery
```
```suggestion
Sphinx-Gallery
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+Sphinx gallery
+==============
+
+This example shows how to add a new example to the PyAnsys `Sphinx-Gallery
+<https://sphinx-gallery.github.io/>`_.
```
```suggestion
This example shows how to add a new example when using `Sphinx-Gallery
<https://sphinx-gallery.github.io/>`_.
```
I rewrote the sentence because the link was to the Sphinx Gallery documentation and had nothing to do with PyAnsys?

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+This example shows how to add a new example to the PyAnsys `Sphinx-Gallery
+<https://sphinx-gallery.github.io/>`_.
+
+To add sphinx-gallery in to your project, you need to install the following packages:
```
```suggestion
To use Sphinx-Gallery, you must first install the package with this command:
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+
+        pip install sphinx-gallery
+
+then add the following to your ``conf.py`` file:
```
```suggestion
Then, add the package to the ``extensions`` variable in your Sphinx ``conf.py`` file:
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+        ]
+"""
+###############################################################################
+# Plotting a simple sphere using pyvista
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```suggestion
# Plot a simple sphere using PyVista
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+###############################################################################
+# Plotting a simple sphere using pyvista
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This example demonstrates how to plot a simple sphere using PyVista.
```
```suggestion
# This code plots a simple sphere using PyVista.
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+sphere.plot()
+
+###############################################################################
+# Plotting a simple sphere using pyvista with a plotter
```
```suggestion
# Plot a simple sphere using PyVista with a plotter
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+plotter.show()
+
+###############################################################################
+# Rending equations using Math
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This example demonstrates how to render equations using Math.
```
```suggestion
# Render equations using IPython ``math``
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# This example shows how to render equations using the IPython ``math`` module.
```

PipKat commented on doc/source/examples/sphinx-gallery/sphinx-gallery.py:
```diff
@@ -0,0 +1,96 @@
+Latex(r"This is a \LaTeX{} equation: $a^2 + b^2 = c^2$")
+
+###############################################################################
+# Rendering a table in markdown
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This is an example to render the table inside the notebook
```
```suggestion
# Render a table in markdown
# ~~~~~~~~~~~~~~~~~~~~~~~~~~
# This is an example to render a table inside the notebook.
```
Heading says a table in markdown and the text says inside a notebook, which is confusing.

PipKat commented on doc/source/getting-started/installation.rst:
```diff
@@ -0,0 +1,28 @@
+Installation
+############
+
+There are multiple sources for installing the latest stable version. These
```
```suggestion
There are multiple sources for installing the latest stable version of the Ansys Sphinx Theme. These
```

PipKat commented on doc/source/index.rst:
```diff
@@ -1,14 +1,42 @@
+################################
 
-.. include:: ../../README.rst
+Version: |version|
```
Why not have the version in the title like our other documentation? Having it here seems odd--and makes it stand out less.

PipKat commented on doc/source/index.rst:
```diff
@@ -1,14 +1,42 @@
-.. include:: ../../README.rst
+Version: |version|
+
+The Ansys Sphinx theme is a custom theme for Sphinx, a documentation generator
+used for creating project documentation from reStructuredText source files.
+
+This theme is specifically tailored for documentation related to Ansys projects.
+It helps ensure consistency in the look and feel of documentation. Various
+useful extensions are included in the theme to make documentation more
+appealing and user-friendly.
```
```suggestion
The Ansys Sphinx Theme is a custom Ansys-branded theme for use with Sphinx, a documentation generator
for creating project documentation from reStructuredText source files.

This theme is specifically tailored for documentation related to Ansys projects
helping to ensure consistency in its look and feel. Various useful extensions
are included in the theme to make documentation more appealing and user-friendly.
```

PipKat commented on doc/source/index.rst:
```diff
@@ -1,14 +1,42 @@
+        :link: getting-started
+        :link-type: doc
+
+        Step-by-step guidelines on how to install the Ansys Sphinx theme.
```
```suggestion
        Learn how to install the Ansys Sphinx Theme.
```

PipKat commented on doc/source/index.rst:
```diff
@@ -1,14 +1,42 @@
+        :link: user-guide
+        :link-type: doc
+
+        Learn about the capabilitie, features, and key topics of this theme.
```
```suggestion
        Learn how to use the capabilities and features of this theme.
```

PipKat commented on doc/source/index.rst:
```diff
@@ -1,14 +1,42 @@
+        :link: examples
+        :link-type: doc
+
+        Gallery of examples showing how third-party extensions integrate with this theme.
```
```suggestion
        Explore examples that show how to integrate third-party extensions with this theme.
```

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+User guide
+##########
+
+This section provides information on how to customize the Ansys Sphinx theme.
```
```suggestion
This section provides information on how to customize the Ansys Sphinx Theme.
```

Revathyvenugopal162 replied:
both i wanted to give information about the basic setting such as switcher , icons and also the customization with third party extensions. so i reformatted as `This section outlines the fundamental configurations of the Ansys Sphinx theme and how to integrate 
third-party extensions with this theme to customize your documentation.`

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+      :link: user-guide/configuration
+      :link-type: doc
+
+      Ansys sphinx theme configuration
```
```suggestion
      Ansys Sphinx Theme configuration
```

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+      :link: user-guide/options
+      :link-type: doc
+
+      Theme options
```
```suggestion
      Set theme options
```

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+      :link: user-guide/autoapi
+      :link-type: doc
+
+      See how to use the autoapi extension with the Ansys sphinx theme
```
```suggestion
      See how to use Sphinx AutoAPI with the Ansys Sphinx Theme
```

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+.. toctree::
+   :maxdepth: 2
+   :hidden:
+   :caption: Ansys sphinx theme configuration
```
```suggestion
   :caption: Ansys Sphinx Theme configuration
```

PipKat commented on doc/source/user-guide.rst:
```diff
@@ -0,0 +1,60 @@
+.. toctree::
+    :maxdepth: 2
+    :hidden:
+    :caption: Ansys sphinx theme extensions
```
```suggestion
    :caption: Ansys Sphinx Theme extensions
```

PipKat commented on doc/source/user-guide/autoapi.rst:
```diff
@@ -1,83 +1,7 @@
-
-``ansys_sphinx_theme_autoapi`` theme options and extension
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Autoapi extension
+==================
```
```suggestion
Sphinx AutoAPI
===============
```

PipKat commented on doc/source/user-guide/autoapi.rst:
```diff
@@ -1,83 +1,7 @@
+Autoapi extension
+==================
 
 To use the ``autoapi`` along with the ``ansys-sphinx-theme``, you need to
```
```suggestion
To use Sphinx AutoAPI with the Ansys Sphinx Theme, you must
```

PipKat commented on doc/source/user-guide/autoapi.rst:
```diff
@@ -1,83 +1,7 @@
+==================
 
 To use the ``autoapi`` along with the ``ansys-sphinx-theme``, you need to
 add ``ansys_sphinx_theme.extension.autoapi`` to the ``extensions`` list in your ``conf.py`` file
```
```suggestion
add ``ansys_sphinx_theme.extension.autoapi`` to the ``extensions`` variable in your Sphinx ``conf.py`` file
```

Revathyvenugopal162 replied:
sure, that sounds good, thanks @PipKat

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+Basic configuration
+===================
+
+Add the following to your `conf.py` file to use the Ansys Sphinx theme:
```
```suggestion
To use the Ansys Sphinx Theme, add the following line to your project's Sphinx ``conf.py`` file:
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+
+   html_theme = "ansys_sphinx_theme"
+
+From ansys sphinx theme you can use the following features:
```
```suggestion
The Ansys Sphinx Theme provides these features:
```
Add a bulleted list that links to each child section. This will be helpful as the text in each section is relatively long.

Revathyvenugopal162 replied:
Sure, added.

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+
+From ansys sphinx theme you can use the following features:
+
+PyAnsys and Ansys logos
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```suggestion
PyAnsys and Ansys logos
~~~~~~~~~~~~~~~~~~~~~~~
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+PyAnsys and Ansys logos
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Ansys Sphinx theme includes the PyAnsys and Ansys logos. All The logos
```
```suggestion
The Ansys Sphinx Theme includes the PyAnsys and Ansys logos. All the logos
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+    html_logo = pyansys_logo_black
+    html_favicon = ansys_favicon
+
+``favicon`` is the icon that appears in the browser tab.
```
```suggestion
The ``favicon`` setting specifies the icon that appears in the browser tab.
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+Version switcher
+~~~~~~~~~~~~~~~~
+
+The Ansys Sphinx theme includes a version switcher that allows users to switch between different versions of the documentation.
+To use the version switcher, add the following code to your `conf.py` file:
```
```suggestion
The Ansys Sphinx Theme includes a version switcher for switching between different versions of the documentation.
To show the version switcher in your documentation, add the following code to your project's Sphinx ``conf.py`` file:
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+       },
+   }
+
+The switcher requires a `versions.json` file that contains the versions of the documentation and their URLs in the given ``json_url``.
+see `PyAnsys multi-version documentation <dev_guide_multi_version_>`_
+for more information.
```
```suggestion
The switcher requires a ``versions.json`` file that contains the versions of the documentation and their URLs in the given ``json_url``.
For more information, see `PyAnsys multi-version documentation <dev_guide_multi_version_>`_ in the
*PyAnsys developer's guide*.
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+PDF cover page
+~~~~~~~~~~~~~~
+
+The Ansys Sphinx theme includes a PDF cover page that you can customize.
```
```suggestion
The Ansys Sphinx Theme includes a PDF cover page that you can customize.
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+Custom CSS
+~~~~~~~~~~
+
+You can add custom CSS to the Ansys Sphinx theme by creating a directory called `_static/css` in
+your documentation and adding the following code to your `conf.py` file:
```
```suggestion
You can add custom CSS to the Ansys Sphinx Theme by creating a directory named ``_static/css`` in
your documentation and adding the following code to your project's Sphinx ``conf.py`` file:
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+   html_static_path = ["_static"]
+   html_css_files = ["css/custom.css"]
+
+Here is an example of a custom CSS file:
```
```suggestion
Here is an example of a custom CSS file that changes the background color
of the body to black and the text color to white:
```

PipKat commented on doc/source/user-guide/configuration.rst:
```diff
@@ -0,0 +1,81 @@
+      color: white;
+   }
+
+which changes the background color of the body to black and the text color to white.
```
```suggestion
```

PipKat commented on doc/source/user-guide/linkcode.rst:
```diff
@@ -1,20 +1,11 @@
-This page contains the shipped extensions that seamlessly integrate with this theme to enhance
-documentation functionality.
-
 Linkcode extension
```
```suggestion
The ``linkcode`` extension
```

PipKat commented on doc/source/user-guide/linkcode.rst:
```diff
@@ -1,20 +1,11 @@
-
 Linkcode extension
--------------------
+===================
```
```suggestion
===========================
```

PipKat commented on doc/source/user-guide/linkcode.rst:
```diff
@@ -1,20 +1,11 @@
--------------------
+===================
 
 The Linkcode extension automatically adds "Source" links to the documentation for Python, C, C++,
 and JavaScript objects. It allows you to link to the source code hosted on GitHub.
```
```suggestion
The Linkcode extension automatically adds "Source" links to the documentation for Python, C, C++,
and JavaScript objects. It allows you to link to the source code hosted on GitHub.
```
```suggestion
The ``linkcode`` extension automatically adds *source* links to the documentation for Python, C, C++,
and JavaScript objects. It allows you to link to the source code hosted on GitHub.
```

PipKat commented on doc/source/user-guide/linkcode.rst:
```diff
@@ -1,20 +1,11 @@
-Installation
-~~~~~~~~~~~~
-
 To use the ``ansys-sphinx-theme`` linkcode extension, you need to add it to the Sphinx configuration:
 
 #. Add the following line in the ``conf.py`` file:
```
```suggestion
To use the ``linkcode`` extension, you must add it to the ``extensions`` variable in your
project's Sphinx ``conf.py`` file:
```
You shouldn't have a procedure with only one step. You will need to fix the indentation of the code block.

PipKat commented on src/ansys_sphinx_theme/extension/autoapi.py:
```diff
@@ -22,6 +22,8 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+"""Module containing an extension for creating autoapi templates provided by theme."""
```
```suggestion
"""Module containing an extension for creating Sphinx AutoAPI templates for the Ansys Sphinx Theme."""
```

PipKat commented on src/ansys_sphinx_theme/extension/linkcode.py:
```diff
@@ -197,8 +197,6 @@ def link_code(app: Sphinx, doctree: Node):
 
-    Notes
-    -----
     Adapted from the original `linkcode` extension in Sphinx, see
```
```suggestion
    Adapted from the original ``linkcode`` extension in Sphinx. For more information, see
```
What is adapted? This sentence needs a sentence and it should have display text for the link (rather than the URL itself).

Revathyvenugopal162 replied:
changed this to
```
The ``linkcode`` extension in this version has been adapted from the original 
`Sphinx linkcode extension <https://github.com/sphinx-doc/sphinx/blob/main/sphinx/ext/linkcode.py>`_.
This adaptation involves modifying the original code and functionality from the Sphinx library.
```

PipKat commented on src/ansys_sphinx_theme/extension/linkcode.py:
```diff
@@ -197,8 +197,6 @@ def link_code(app: Sphinx, doctree: Node):
     The `link_code_library` configuration option can be used to specify a custom library
     for link resolution.
```
```suggestion
    You can use the ``link_code_library`` configuration option to specify a custom library
    for link resolution.
```
I'd like to make some edits to the preceding lines but I can't do so in this PR. There are additional places where single backticks are used instead of double backticks. Single backticks render as italics in the doc (rather than as a code entity).

PipKat reviewed (commented):
Lots of suggestions for you.

Revathyvenugopal162 commented:
Thank you for the detailed review, @PipKat. I have addressed all the comments,  I hope you can open a documentation review at your convenience once this PR is merged as mentioned above.
